Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1141,14 +1141,113 @@ open class NavigationMapView: UIView {
+        }
+        let routesContainTolls = tollRoutes.count > 0
+        
+        // Pick a random tail direction to keep things varied.
+        guard let randomTailPosition = [
+            RouteDurationAnnotationTailPosition.leading,
+            RouteDurationAnnotationTailPosition.trailing
+        ].randomElement() else { return }
+
+        var features = [Turf.Feature]()
+        
+        for (index, alternativeRoute) in routes.enumerated() {
+            guard let routeShape = alternativeRoute.indexedRouteResponse.currentRoute?.shape else { return }
+            
+            var annotationCoordinate: LocationCoordinate2D!
+            if let mainRoute = self.routes?.first {
+                let offset = mainRoute.distance - alternativeRoute.infoFromDeviationPoint.distance + continuousAlternativeDurationAnnotationOffset
+                annotationCoordinate = routeShape.indexedCoordinateFromStart(distance: offset)?.coordinate
+            } else {
+                annotationCoordinate = routeShape.indexedCoordinateFromStart(distance: alternativeRoute.infoFromOrigin.distance - alternativeRoute.infoFromDeviationPoint.distance + continuousAlternativeDurationAnnotationOffset)?.coordinate
+            }
+            
+            // Form the appropriate text string for the annotation.
+            let labelText = self.annotationLabelForAlternativeRoute(alternativeRoute,
+                                                                    tolls: routesContainTolls)
+            
+            // Create the feature for this route annotation. Set the styling attributes that will be
+            // used to render the annotation in the style layer.
+            var feature = Feature(geometry: .point(Point(annotationCoordinate)))
+            
+            var tailPosition = randomTailPosition
+            
+            // Convert our coordinate to screen space so we can make a choice on which side of the
+            // coordinate the label ends up on.
+            let unprojectedCoordinate = mapView.mapboxMap.point(for: annotationCoordinate)
+            
+            // Pick the orientation of the bubble "stem" based on how close to the edge of the screen it is.
+            if tailPosition == .leading && unprojectedCoordinate.x > bounds.width * 0.75 {
+                tailPosition = .trailing
+            } else if tailPosition == .trailing && unprojectedCoordinate.x < bounds.width * 0.25 {
+                tailPosition = .leading
+            }
+            
+            let imageName = tailPosition == .leading ? "RouteInfoAnnotationLeftHanded" : "RouteInfoAnnotationRightHanded"
+            
+            // Set the feature attributes which will be used in styling the symbol style layer.
+            feature.properties = [
+                "selected": .boolean(false),
+                "tailPosition": .number(Double(tailPosition.rawValue)),
+                "text": .string(labelText),
+                "imageName": .string(imageName),
```

Comment:
Unfortunately it does. I didn't check what might be the cause

Reply:
Let’s ticket that out as tail work for both features then. At a glance, I don’t see what would cause the issue; maybe something to do with how the image asset is stretched: mapbox/mapbox-maps-ios#381.